### DISCUSSION TITLE
Enable External I/O signal for relay on user selectable battery charge thresholds.

### DISCUSSION THREAD

**Discussion by 2Fblob** [syssi/esphome-jk-bms#626] [Q&A]
Is there a way to program an external relay to come on / off at a custom battery level, before the BMS cuts out?

Thanks!

**Comment by syssi** [comment]
The JK-BMS doesn't provide this feature. If you are able to predict the error/protection you could react by adding some additional logic to the YAML.

**Reply by 2Fblob** [reply]
The battery capacity and total voltage are given and passed on to home assistant - so I thought it would be possible for the ESP32 to directly process these, with some coding on the YAML. I have no experience programming in YAML, unfortunately - Only C.

**Reply by syssi** [reply]
Give it a try: https://esphome.io/components/sensor/index.html#on-value-range
